From 04c3622dd853b62078feb92b7e2f05e09f53e755 Mon Sep 17 00:00:00 2001
From: Giulio Fidente <gfidente@redhat.com>
Date: Tue, 23 Jun 2015 14:21:53 +0200
Subject: [PATCH] Remove control over the galera_master_node

We do not want to give users control over the galera_master_node,
this should be gathered using the clustercheck script instead.

Depends-On: I56ebd2d8405ac35c707666d993b396f04aeb683e
Change-Id: Ib6a36e9283b73133251fb9ff3f33e71c50edb3db
Closes-Bug: 1467918
Closes-Bug: https://bugzilla.redhat.com/show_bug.cgi?id=1234817
---
 tripleo/manifests/loadbalancer.pp | 25 ++-----------------------
 1 file changed, 2 insertions(+), 23 deletions(-)

diff --git a/tripleo/manifests/loadbalancer.pp b/tripleo/manifests/loadbalancer.pp
index e94f44a..ac001fd 100644
--- a/tripleo/manifests/loadbalancer.pp
+++ b/tripleo/manifests/loadbalancer.pp
@@ -132,14 +132,6 @@
 #  When set, enables SSL on the Horizon public API endpoint using the specified file.
 #  Defaults to undef
 #
-# [*galera_master_hostname*]
-#  FQDN of the Galera master node
-#  Defaults to undef
-#
-# [*galera_master_ip*]
-#  IP of the Galera master node
-#  Defaults to undef
-#
 # [*keystone_admin*]
 #  (optional) Enable or not Keystone Admin API binding
 #  Defaults to false
@@ -240,8 +232,6 @@ class tripleo::loadbalancer (
   $swift_certificate         = undef,
   $heat_certificate          = undef,
   $horizon_certificate       = undef,
-  $galera_master_hostname    = undef,
-  $galera_master_ip          = undef,
   $keystone_admin            = false,
   $keystone_public           = false,
   $neutron                   = false,
@@ -874,22 +864,11 @@ class tripleo::loadbalancer (
       },
       collect_exported => false,
     }
-
-    haproxy::balancermember { 'mysql':
-      listening_service => 'mysql',
-      ports             => '3306',
-      ipaddresses       => $galera_master_ip,
-      server_names      => $galera_master_hostname,
-      options           => ['check', 'inter 2000', 'rise 2', 'fall 5'],
-    }
-
-    $controller_hosts_without_galera_master = delete(hiera('mysql_node_ips', $controller_hosts_real), $galera_master_ip)
-    $controller_hosts_names_without_galera_master = delete($controller_hosts_names_real, downcase($galera_master_hostname))
     haproxy::balancermember { 'mysql-backup':
       listening_service => 'mysql',
       ports             => '3306',
-      ipaddresses       => $controller_hosts_without_galera_master,
-      server_names      => $controller_hosts_names_without_galera_master,
+      ipaddresses       => hiera('mysql_node_ips', $controller_hosts_real),
+      server_names      => $controller_hosts_names_real,
       options           => ['check', 'inter 2000', 'rise 2', 'fall 5', 'backup'],
     }
   }
